.circleci/config.yml

```diff
@@ -312,6 +313,75 @@ jobs:
     <<: *test-defaults
     environment:
       - TEST_TARGET=sanity
+  build-waterfall:
+    <<: *defaults
+    working_directory: ~/
+    steps:
+      - checkout:
+          path: emscripten/
+      - run:
+          name: install package dependencies
+          command: |
+            apt-get update -q
+            apt-get install -q -y python3 cmake
+      - run:
+          name: install emsdk
+          command: |
+            wget https://github.com/juj/emsdk/archive/master.tar.gz
+            tar -xf master.tar.gz
+            cd emsdk-master
+            ./emsdk --notty update-tags
+            export WATERFALL_LKGR=`python -c "print [x.split('\"')[3] for x in open('upstream/lkgr.json').readlines() if '\"build\"' in x][0]"`
+            echo "LKGR: $WATERFALL_LKGR"
+            ./emsdk --notty install node-8.9.1-64bit # TODO: should emsdk install this automatically?
+            ./emsdk --notty install clang-upstream-$WATERFALL_LKGR-64bit
+            ./emsdk --notty activate clang-upstream-$WATERFALL_LKGR-64bit
+            cd -
+            # Use LLVM, clang, lld binaries from the emsdk
+            export LLVM="$HOME/emsdk-master/upstream/$WATERFALL_LKGR/bin"
+            # Remove the emsdk version of emscripten to save space in the
+            # persistent workspace and to avoid any confusion with that version
+            # we are trying to test.
+            rm -r ~/emsdk-master/upstream/$WATERFALL_LKGR/emscripten/
+            # Use Binaryen from the emsdk's waterfall (wasm backend can't use ports)
+            echo BINARYEN_ROOT="'$HOME/emsdk-master/upstream/$WATERFALL_LKGR/'" >> ~/.emscripten
+            # FIXME fix node
+            echo NODE_JS="'$HOME/emsdk-master/node/8.9.1_64bit/bin/node'" >> ~/.emscripten
+            echo "COMPILER_ENGINE = NODE_JS" >> ~/.emscripten
+            echo "JS_ENGINES = [NODE_JS]" >> ~/.emscripten
+            echo "==="
+            cat ~/.emscripten
+            echo "==="
```

Comment:
Probably don't need this debugging in the final version?

Reply:
Might be useful enough to clear up any assumptions that my instinct says leave it.
Having more info than usual on CI makes debugging it much easier.
Not that I expect the `.emscripten` specifically to cause too much fuss, but in general.

Reply:
Fixing, thanks.

Reply:
Oops, just saw @jgravelle-google's comment after I wrote mine.

I agree in general that debugging can be useful on CI, but I admit this particular one was mainly useful in bringing it up, and likely won't help much later.